### PR TITLE
Lint Clean Up

### PR DESCRIPTION
**Pull request by MMurrayAtBR**
Updating of files to pass lint checker and removal of unnecessary Suppress annotations

### CODE REVIEW

File: app/src/main/res/drawable/ic_home.xml [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
-  <group>
-    <clip-path
-        android:pathData="M2,2.445h19.484L21.484,22.96L2,22.96z"/>
-    <path
-        android:pathData="M11.743,2.445L2,10.273v12.503l19.484,0.184L21.484,10.234L11.743,2.444zM11.746,5.315l7.498,5.997v9.386l-15.004,-0.14L4.24,11.345l7.506,-6.03z"
-        android:fillColor="#E2231A"
-        android:fillType="nonZero"/>
-  </group>
+    <!-- "clip-path" attribute not supported API < 24; Check that removing
+    this path data does not affect the look of the graphic -->
```
Comment (jaredstockton) [inline app/src/main/res/drawable/ic_home.xml]: @MMurrayAtBR This comment can be removed now that the minSdkVersion is set to 24

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 2]
```diff
@@ -89,7 +89,7 @@ object Config {
 
         // https://developer.android.com/studio/releases/build-tools
         const val BUILD_TOOLS = "31.0.0"
-        const val MIN_SDK = 23 // TODO: TEMPLATE - Replace with appropriate project minSdkVersion
+        const val MIN_SDK = 24 // TODO: TEMPLATE - Replace with appropriate project minSdkVersion
```
Comment (jaredstockton) [inline buildSrc/src/main/kotlin/Dependencies.kt:92]: @MMurrayAtBR While running `lintInternalDebug` locally, it was mentioned that there was a `drawable-v24` resource . That resource should actually be deleted as it uses an incorrect app icon for v24+ devices. The existing resource in `drawable` is the preferred drawable.

- [x] Delete `ic_launcher_foreground.xml` from `drawable-v24`

File: compose/src/main/java/com/bottlerocketstudios/compose/auth/AuthCode.kt [hunk 3]
```diff
@@ -63,6 +63,8 @@ fun AuthCodePreview() {
     }
 }
 
+// FIX ME: javaScriptEnabled = true can result in removal from Play Store
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/auth/AuthCode.kt]: @MMurrayAtBR nit: Change `FIX ME` `FIXME` to follow the naming used elsewhere in the project. Additionally, using `FIXME` allows project-wide searches to show all instances as well as cause all instances to show in the TODO view of Android Studio.

```suggestion
// FIXME: javaScriptEnabled = true can result in removal from Play Store
```

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt [hunk 4]
```diff
@@ -14,12 +13,12 @@ object Routes {
     const val CreateSnippet = "snippets/create"
     const val Profile = "profile"
 
-    fun RepositoryBrowser(data: RepositoryBrowserData) =
+    fun repositoryBrowser(data: RepositoryBrowserData) =
```
Comment (jaredstockton) [inline app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt]: @BottleRocket-Colin I understand why you used UpperCamelCase for these function names (mimicking the approach use for Composable functions). I don't really have a strong opinion for or against them so would appreciate to hear your feedback. Have you seen examples of Google / other sample repositories using UpperCamelCase style for routes?

cc: @MMurrayAtBR

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -19,4 +19,18 @@
     <!-- Misc -->
     <string name="dev_options_button">Dev Options</string>
 
+    <string name="today">Today</string>
```
Comment (jaredstockton) [inline app/src/main/res/values/strings.xml:22]: @MMurrayAtBR Why are these necessary in the app module?

File: compose/src/main/java/com/bottlerocketstudios/compose/navdrawer/NavDrawer.kt [hunk 6]
```diff
@@ -124,24 +124,28 @@ fun ProfileScreenPreview() {
...
-        ) {},
-        NavItemState(
-            icon = R.drawable.ic_hide,
-            itemText = R.string.login_id_hint,
-            selected = false
-        ) {}
-    )
-)
+fun generateItemListPreview(): List<NavItemState> {
+    val itemList by remember {
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/navdrawer/NavDrawer.kt:129]: @MMurrayAtBR This approach seems fine to me.

@BottleRocket-Colin Thoughts?

**Review by jaredstockton — COMMENTED** [review]
@MMurrayAtBR Thanks for the PR! I have left a few comments to address prior to approval :)

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt (hunk 4 above)
Comment (BottleRocket-Colin) [reply app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt]: Good Catch!.  @MMurrayAtBR Let's change these back to UpperCamelCase and add a suppress annotation.  That way they will be consistent with the routes above.

File: compose/src/main/java/com/bottlerocketstudios/compose/navdrawer/NavDrawer.kt (hunk 6 above)
Comment (BottleRocket-Colin) [reply compose/src/main/java/com/bottlerocketstudios/compose/navdrawer/NavDrawer.kt:129]: Works for me.

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (MMurrayAtBR) [reply app/src/main/res/values/strings.xml:22]: These string resources are being used in the app module "testResources" & "testContext" mocks, but were not in the app module string resources. 
I chose to add the missing strings to the app module res file & resolve the Unresolved References, instead of making changes to the mocks

**Review by jaredstockton — APPROVED** [review]
@MMurrayAtBR Thanks for making the changes! Looks good now :)

**Review by BottleRocket-Colin — APPROVED** [review]
